packages/core/src/components/axes/axis.ts

```diff
@@ -70,13 +70,44 @@ export class Axis extends Component {
 				break;
 		}
 
+		// Add axis into the parent
+		const container = DOMUtils.appendOrSelect(svg, `g.axis.${axisPosition}`);
+		const axisRefExists = !container.select(`g.ticks`).empty();
+		let axisRef = DOMUtils.appendOrSelect(container, `g.ticks`);
+		if (!axisRefExists) {
+			axisRef.attr("role", `${Roles.GRAPHICS_OBJECT} ${Roles.GROUP}`);
+		}
+
+		// We draw the invisible axis because of the async nature of d3 transitions
+		// To be able to tell the final width & height of the axis when initiaing the transition
+		// The invisible axis is updated instantly and without a transition
+		const invisibleAxisRef = DOMUtils.appendOrSelect(container, `g.ticks.invisible`)
+			.style("opacity", "0")
+			.attr("aria-hidden", true);
+
+		// Get font height
+		const fakeTick = DOMUtils.appendOrSelect(invisibleAxisRef, `g.tick`);
+		const fakeTickText = DOMUtils.appendOrSelect(fakeTick, `text`).text("0");
+		const tickHeight = fakeTickText.node().getBBox().height;
+		fakeTick.remove();
+
 		// Initialize axis object
 		const axis = axisFunction(scale)
 			.tickSizeOuter(0)
 			.tickFormat(Tools.getProperty(axisOptions, "ticks", "formatter"));
 
 		if (scale.ticks) {
-			const numberOfTicks = Tools.getProperty(axisOptions, "ticks", "number") || Configuration.axis.ticks.number;
+			let numberOfTicks = Tools.getProperty(axisOptions, "ticks", "number");
+
+			if (axisPosition === AxisPositions.LEFT || axisPosition === AxisPositions.RIGHT) {
+				if (!numberOfTicks) {
+					const numberOfTicksThoretical = Math.floor(height / (tickHeight * 2.5));
+					numberOfTicks = Tools.clamp(numberOfTicksThoretical, 2, Configuration.axis.ticks.number);
+				}
+			} else {
+				numberOfTicks = numberOfTicks || Configuration.axis.ticks.number;
+			}
```

Comment:
What if I move the y-axis on the left to the bottom? (e.g. horizontal bar)

We should really have a way of identifying the nature of the scale domain and based on that decide whether we should remove some ticks

Reply:
What we are trying to fix with this PR is to adjust the number of ticks on vertical axises based on the chart height.

We are not interested on whether it's the Y or X axis, we only care if it is a vertical one.
Do you think it's the wrong logic?

We can extend this solution to horizontal axes in the future too.